File: oop/heritage/exercice4/src/Menuisier.java


public class Menuisier extends Personne {
    private String metier;

    public Menuisier() {
        super();
    }

    public Menuisier(String prenom) {
        super(prenom);
        metier = "Menuisier";
    }

    @Override
    public void affiche() {
        super.affiche();
        System.out.print(" le " + this.metier);
    }
}

File: oop/heritage/CPP/exercice4/src/Menuisier.cpp
#include "Menuisier.hpp"

Menuisier::Menuisier() : Personne(), metier("") {}

Menuisier::Menuisier(const std::string &prenom) : Personne(prenom), metier("Menuisier") {}

void Menuisier::affiche() const
{
    Personne::affiche();
    std::cout << " le " << metier;
}
